source/js/storage.js

```diff
@@ -0,0 +1,189 @@
+const userInfo = {};
+const renderedText = {};
+
+window.addEventListener("DOMContentLoaded", init);
+
+// temporary init for demo
+async function init() {
+  getUserInfo().then(renderText);
+  bindButton();
+}
+
+/**
+ * After this function resolves, the userInfo object
+ * should be populated with the userInfo taken from storage
+ * @async
+ * @function
+ * @returns {Promise}
+ */
+async function getUserInfo() {
+  return new Promise((resolve, reject) => {
+    // attempt to retrieve
+    try {
+      let storageUserInfoString = window.localStorage.getItem("userInfo");
+      let storageUserInfo = {};
+      // if it doesn't exist, initialize it as a blank user;
+      if (!storageUserInfoString) {
+        storageUserInfo = {
+          savedRecipes: [],
+        };
+        window.localStorage.setItem(
+          "userInfo",
+          JSON.stringify(storageUserInfo)
+        );
+      } else {
+        storageUserInfo = JSON.parse(storageUserInfoString);
+      }
+
+      // update global variable
+      for (let attr in storageUserInfo) {
+        userInfo[attr] = storageUserInfo[attr];
+      }
+      // all done!
+      resolve(true);
+    } catch (error) {
+      // uh oh
+      console.log("Unable to retrieve userInfo", error);
```

Comment:
Codacy has a fix for the issue: [There should be a line break after this argument.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8429666)
```suggestion
      console.log("Unable to retrieve userInfo",
```